services/src/main/java/org/fao/geonet/api/records/InspireValidationApi.java

```diff
@@ -0,0 +1,237 @@
+import jeeves.services.ReadWriteController;
+import springfox.documentation.annotations.ApiIgnore;
+
+@RequestMapping(value = {
+        "/api/inspire",
+        "/api/" + API.VERSION_0_1 +
+        "/inspire"
+})
+@Api(value = API_CLASS_RECORD_TAG,
+tags = API_CLASS_RECORD_TAG)
+@Controller("inspire")
+@PreAuthorize("hasRole('Editor')")
+@ReadWriteController
+public class InspireValidationApi {
+
+    @ApiOperation(
+            value = "Submit a record to the INSPIRE service for validation.",
+            notes = "User MUST be able to edit the record to validate it. "
+                    + "An INSPIRE endpoint must be configured in Settings. "
+                    + "This activates an asyncronous process, this method does not return any report. "
+                    + "This method returns an id to be used to get the report.",
+                    nickname = "submitValidate")
+    @RequestMapping(value = "/{metadataUuid}/validate/submit",
+    method = RequestMethod.PUT,
+    produces = {
+            MediaType.TEXT_PLAIN_VALUE
+    }
+            )
+    @PreAuthorize("hasRole('Editor')")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Check status of the report."),
+            @ApiResponse(code = 404, message = "Metdata not found."),
+            @ApiResponse(code = 500, message = "Service unavailable."),
+            @ApiResponse(code = 403, message = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_EDIT)
+    })
+    public
+    @ResponseBody
+    String validateRecord(
+            @ApiParam(
+                    value = API_PARAM_RECORD_UUID,
+                    required = true)
+            @PathVariable
+            String metadataUuid,
+            HttpServletResponse response,
+            @ApiParam(hidden = true)
+            @ApiIgnore
+            HttpServletRequest request,
+            @ApiParam(hidden = true)
+            @ApiIgnore
+            HttpSession session
+            ) throws Exception {
+
+        ApplicationContext appContext = ApplicationContextHolder.get();
+        final SettingManager settingManager = appContext.getBean(SettingManager.class);
+        Metadata metadata = ApiUtils.canEditRecord(metadataUuid, request);
+
+        if(metadata==null) {
+            response.setStatus(HttpStatus.SC_NOT_FOUND);
+            return "";
+        }
+
+        if(!metadata.getDataInfo().getSchemaId().startsWith("iso19139")) {
+            response.setStatus(HttpStatus.SC_NOT_ACCEPTABLE);
+            return "";
+        }
+
+        String id = String.valueOf(metadata.getId());
+
+        String URL = settingManager.getValue(Settings.SYSTEM_INSPIRE_REMOTE_VALIDATION_URL);
+
+        try {
+            Element md = (Element) ApiUtils.getUserSession(session).getProperty(Geonet.Session.METADATA_EDITING + id);
+            if (md == null) {
+                throw new ResourceNotFoundException(String.format("Requested metadata with id '%s' is not available in current session. "
```

Comment:
Check to move the message to message bundle files. Not sure if done in other services, but should be good to use them to provide localised error messages.